modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -1035,6 +1092,12 @@ private void exec() {
                     k = operands[2];
                     sf.intRegs[k] = sf.refRegs[i] == sf.refRegs[j] ? 1 : 0;
                     break;
+                case InstructionCodes.TEQ:
+                    i = operands[0];
+                    j = operands[1];
+                    k = operands[2];
+                    sf.intRegs[k] = sf.refRegs[i].value() == sf.refRegs[j].value() ? 1 : 0;
```

Comment:
+1 @sameera-jayasoma , since we are using singleton instances of types, Java object comparison works for these, even for array types. How for arrays of arrays cases, internal element type does not resolve to singleton type ( internal element type it self can be an array type  ). Added necessary logic to handle. Can you please check?